src/pseudopeople/noise_functions.py

```diff
@@ -46,6 +43,80 @@ def omit_rows(
     return noised_data
 
 
+def _get_census_omission_noise_levels(
+    population: pd.DataFrame,
+    base_probability: float = data_values.DO_NOT_RESPOND_BASE_PROBABILITY,
+) -> pd.Series:
+    """
+    Helper function for do_not_respond noising based on demography of age, race/ethnicity, and sex.
+
+    :param population: a dataset containing records of simulants
+    :param base_probability: base probability for do_not_respond
+    :return: a pd.Series of probabilities
+    """
+    probabilities = pd.Series(base_probability, index=population.index)
+    probabilities += (
+        population["race_ethnicity"]
+        .astype(str)
+        .map(data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_RACE)
+    )
+    for sex in ["Female", "Male"]:
+        sex_mask = population["sex"] == sex
+        age_bins = pd.cut(
+            x=population[sex_mask]["age"],
+            bins=data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE[sex].index,
+        )
+        probabilities[sex_mask] += age_bins.map(
+            data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE[sex]
+        ).astype(float)
+    probabilities[probabilities < 0.0] = 0.0
```

Comment:
Is it possible for the probability to be greater than 1? Maybe add `probabilities[probabilities > 1.0] = 1.0` just to be safe?

Reply:
I think it's very unlikely, maybe impossible with the current input data. Not a bad idea to add the check you are suggesting.